proxy/src/bind.rs

```diff
@@ -50,7 +51,7 @@ pub struct BindProtocol<C, B> {
 /// that each host receives its own connection.
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub enum Protocol {
-    Http1(Host),
+    Http1 { host: Host, was_absolute_form: bool },
```

Comment:
Hyper changed the setting of whether or not to use absolute URIs from `Request` to the `Connect` return type. Therefore, we have to detect whether or not the request is absolute form here instead of where we did so previously.